File: bot/src/main/java/net/slimediamond/atom/discord/commands/PingCommand.java
package net.slimediamond.atom.discord.commands;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.slimediamond.atom.command.discord.DiscordCommandContext;
import net.slimediamond.atom.command.discord.DiscordCommandExecutor;

import java.awt.*;

public class PingCommand implements DiscordCommandExecutor {
    @Override
    public void execute(DiscordCommandContext context) throws Exception {
        MessageEmbed embed = new EmbedBuilder()
                .setColor(Color.decode("#4feb34"))
                .setTitle("Pong!")
                .setDescription("Latency: " + context.getJDA().getGatewayPing() + "ms")
                .build();

        context.replyEmbeds(embed);
    }
}

File: bot/src/main/kotlin/net/slimediamond/atom/commands/PingCommand.kt
package net.slimediamond.atom.commands

import net.slimediamond.atom.api.command.CommandNode
import net.slimediamond.atom.api.command.CommandNodeContext
import net.slimediamond.atom.api.command.CommandResult

class PingCommand : CommandNode("Replies with pong", "ping") {

    override suspend fun execute(context: CommandNodeContext): CommandResult {
        context.sendMessage("Pong")
        return CommandResult.success
    }

}